### ISSUE TITLE
Serve DLNA not working properly with OneDrive for Business

### ISSUE THREAD

**Issue by dxzdxz1** [rclone/rclone#7259]
### What is the problem you are having with rclone?

`rclone serve dlna` is giving me the 'unrecognized file type' error with some files.

### What is your rclone version (output from `rclone version`)

rclone v1.63.1
and also
rclone v1.64.0-beta.7299.11eeaaf79

### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

- os/version: Microsoft Windows 11 Pro 23H2 (64 bit)
- os/kernel: 10.0.22631.2215 (x86_64)

### Which cloud storage system are you using? (e.g. Google Drive)

OneDrive for Business

### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
cd /d "%~dp0"
start rclone serve dlna obusiness:/Videos/test ^
--config=rclone.conf ^
--vfs-cache-mode writes ^
--vfs-read-chunk-size-limit 1G ^
--read-only ^
--cache-dir=appdata\cache ^
--log-level=DEBUG
```

### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
2023/08/30 11:46:25 DEBUG : Creating backend with remote "obusiness:/Videos/test"
2023/08/30 11:46:25 DEBUG : Using config file from "D:\\Apps\\rClone\\rclone.conf"
2023/08/30 11:46:26 DEBUG : fs cache: renaming cache item "obusiness:/Videos/test" to be canonical "obusiness:Videos/test"
2023/08/30 11:46:26 DEBUG : OneDrive root 'Videos/test': Next delta token is: 
2023/08/30 11:46:26 DEBUG : vfs cache: root is "D:\\Apps\\rClone\\appdata\\cache"
2023/08/30 11:46:26 DEBUG : vfs cache: data root is "\\\\?\\D:\\Apps\\rClone\\appdata\\cache\\vfs\\obusiness\\Videos\\test"
2023/08/30 11:46:26 DEBUG : vfs cache: metadata root is "\\\\?\\D:\\Apps\\rClone\\appdata\\cache\\vfsMeta\\obusiness\\Videos\\test"
2023/08/30 11:46:26 DEBUG : Creating backend with remote "D:/Apps/rClone/appdata/cache/vfs/obusiness/Videos/test"
2023/08/30 11:46:26 DEBUG : fs cache: renaming cache item "D:/Apps/rClone/appdata/cache/vfs/obusiness/Videos/test" to be canonical "//?/D:/Apps/rClone/appdata/cache/vfs/obusiness/Videos/test"
2023/08/30 11:46:26 DEBUG : Creating backend with remote "D:/Apps/rClone/appdata/cache/vfsMeta/obusiness/Videos/test"
2023/08/30 11:46:26 DEBUG : fs cache: renaming cache item "D:/Apps/rClone/appdata/cache/vfsMeta/obusiness/Videos/test" to be canonical "//?/D:/Apps/rClone/appdata/cache/vfsMeta/obusiness/Videos/test"
2023/08/30 11:46:26 INFO  : vfs cache: cleaned: objects 0 (was 0) in use 0, to upload 0, uploading 0, total size 0 (was 0)
2023/08/30 11:46:26 NOTICE: OneDrive root 'Videos/test': Serving HTTP on 0.0.0.0:7879
2023/08/30 11:46:26 NOTICE: DLNA server on :7879: Started SSDP on Ethernet
2023/08/30 11:46:38 INFO  : /static/ContentDirectory.xml: 192.168.15.3:60226 GET 200
2023/08/30 11:46:38 INFO  : /static/ContentDirectory.xml: 192.168.15.3:60224 GET 200
2023/08/30 11:46:38 INFO  : /static/ConnectionManager.xml: 192.168.15.3:60226 GET 200
2023/08/30 11:46:38 INFO  : /static/ConnectionManager.xml: 192.168.15.3:60224 GET 200
2023/08/30 11:46:38 INFO  : /static/X_MS_MediaReceiverRegistrar.xml: 192.168.15.3:60226 GET 200
2023/08/30 11:46:38 INFO  : /static/X_MS_MediaReceiverRegistrar.xml: 192.168.15.3:60224 GET 200
2023/08/30 11:46:39 ERROR : /: 192.168.15.3:60226 SUBSCRIBE 404
2023/08/30 11:46:39 ERROR : /: 192.168.15.3:60224 SUBSCRIBE 404
2023/08/30 11:46:39 ERROR : /: 192.168.15.3:60230 SUBSCRIBE 404
2023/08/30 11:46:39 ERROR : /: 192.168.15.3:60232 SUBSCRIBE 404
2023/08/30 11:46:43 DEBUG : DLNA server on :7879: unrecognized file type: 10.mkv
2023/08/30 11:46:43 DEBUG : DLNA server on :7879: unrecognized file type: 2.ts
2023/08/30 11:46:43 DEBUG : DLNA server on :7879: unrecognized file type: 5.3gp
2023/08/30 11:46:43 DEBUG : DLNA server on :7879: unrecognized file type: 6.vob
```

I noticed that most of my files weren't showing on my TV, so I create a test scenario to try to narrow down what's the problem.
I created a folder with 10 files that my TV and my phone can read through DLNA: mkv, mp4, ts, avi, mpg, vob, wmv, webm, mov and 3gp. From these 10 files, 4 are being ignored on both my TV and on VLC for Android: mkv, ts, vob, 3gp

I've only tested with video files so far, but the problem seems to be the way rclone parses the files from onedrive for business or a problem with onedrive for business itself. I have the same files on my OneDrive personal account, and all of them are playing if I start `rclone serve dlna` using my OneDrive personal config.

For instance, if i use `rclone lsjson` to get the info from the files that aren't showing on OneDrive for Business, it shows the following:

` "Path":"2.ts","Name":"2.ts","Size":1332695387,"MimeType":"application/octet-stream" `
`"Path":"10.mkv","Name":"10.mkv","Size":1331362846,"MimeType":"application/octet-stream"`

And if I use `rclone lsjson` for the same file on my personal account, it shows:

` "Path":"2.ts","Name":"2.ts","Size":1332695387,"MimeType":"video/MP2T" `
`"Path":"10.mkv","Name":"10.mkv","Size":1331362846,"MimeType":"video/x-matroska"`

Is this a problem with rclone or with onedrive? If it's a problem with Onedrive, is there a way to fix using rclone?

I can also do more tests and provide more logs if necessary. Thanks
‎‎‎‎‎‎‎‎ㅤ
‎‎‎‎‎‎‎‎ㅤ
‎‎‎‎‎‎‎‎ㅤ
‎‎‎‎‎‎‎‎ㅤ
‎‎‎‎‎‎‎‎ㅤ

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by dxzdxz1** [comment]
> The problem appears to be the `MimeType` or `Content-Type` of the files are wrong on OneDrive Business - is that correct?
> 
> So I think this is caused by the OneDrive server not understanding the type of file.

Yes, it makes a lot of sense

Being an issue with OneDrive for Business itself, I imagine it's not possible to override the mimetype value locally directly on rclone  right before send the files info to DLNA devices, right? I was reading the docs and didn't find nothing that covers mimetype changes.

Oh, another thing I tried was:

First I started a `rclone serve dlna` session using my personal account, opened the dlna app on my TV, waited for rclone to list all my files and then a closed the rclone session from my personal account and started a new session with the OneDrive for Business account.
My TV app didn't reload the info, and since my files are located on the same path on both accounts, my TV was able to start a stream with OneDrive for business. Funny, right? Is this info useful somehow?

**Comment by ncw** [comment]
> Being an issue with OneDrive for Business itself, I imagine it's not possible to override the mimetype value locally directly on rclone right before send the files info to DLNA devices, right?

That is what I was looking in to. I don't think it is possible unfortunately.

Try this workaround which looks up the content type from the extension if the backend thinks it is `application/octet-stream`.

[v1.64.0-beta.7303.a82343446.fix-7259-serve-dlna-mime-type](https://beta.rclone.org/branch/fix-7259-serve-dlna-mime-type/v1.64.0-beta.7303.a82343446.fix-7259-serve-dlna-mime-type/) on branch [fix-7259-serve-dlna-mime-type](https://github.com/rclone/rclone/tree/fix-7259-serve-dlna-mime-type) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
Thank you! And thanks for testing.

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org/) in 15-30 minutes and released in v1.64
